modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/IgniteMdCollation.java

```diff
@@ -318,7 +319,7 @@ private IgniteMdCollation() {
             if (project.e instanceof RexInputRef) {
                 targets.compute(((RexSlot) project.e).getIndex(), (k, v) -> {
                             if (v == null) {
-                                return new ArrayList<>(Collections.singleton(project.i));
+                                return new IntArrayList(Collections.singleton(project.i));
```

Comment:
The main goal of using IntList (avoid boxing/unboxing) is still not achieved. 
```suggestion
                                return new IntArrayList(project.i);
```